File: programming-languages/CPlusPlus/full-example.cpp
#include <iostream>
#include <string>

// Função que verifica se a pessoa é maior de idade
bool maiorDeIdade(int idade) {
    return idade >= 18;
}

int main() {
    std::string nome;
    int idade;

    std::cout << "Digite seu nome: ";
    std::getline(std::cin, nome);

    std::cout << "Digite sua idade: ";
    std::cin >> idade;

    if (maiorDeIdade(idade)) {
        std::cout << nome << " é maior de idade." << std::endl;
    } else {
        std::cout << nome << " é menor de idade." << std::endl;
    }

    return 0;
}

File: programming-languages/CSharp/full-example.cs
using System;

class Program {
    static bool MaiorDeIdade(int idade) {
        return idade >= 18;
    }

    static void Main() {
        Console.Write("Digite seu nome: ");
        string nome = Console.ReadLine();

        Console.Write("Digite sua idade: ");
        int idade = int.Parse(Console.ReadLine());

        if (MaiorDeIdade(idade)) {
            Console.WriteLine($"{nome} é maior de idade.");
        } else {
            Console.WriteLine($"{nome} é menor de idade.");
        }
    }
}
